app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -1006,16 +992,17 @@ protected ArtifactMetaDataDto getArtifactMetaDataRaw(Handle handle, String group
     }
 
     /**
-     * IMPORTANT: Private methods can't be @Transactional. Callers MUST have started a transaction.
-     *
      * @param references may be null
      */
-    private String getContentHash(String groupId, String artifactId, boolean canonical, TypedContent content,
-            List<ArtifactReferenceDto> references) {
+    private String getContentHash(Handle handle, String groupId, String artifactId, boolean canonical,
+            TypedContent content, List<ArtifactReferenceDto> references) {
         if (canonical) {
-            var artifactMetaData = getArtifactMetaData(groupId, artifactId);
+            var artifactMetaData = getArtifactMetaDataRaw(handle, groupId, artifactId);
+            Function<List<ArtifactReferenceDto>, Map<String, TypedContent>> referenceResolver = (refs) -> {
```

Comment:
Duplicate lambda.

Reply:
If you mean because this lamba is the same as another one in this same class, it's because they need to be local in order to pass the local `Handle` reference.